web/src/pages/feed.tsx

```diff
@@ -43,20 +45,12 @@ export default function Feed({ searchValue }: Props) {
 
   // Debounce the search value (don't immediately update it after every keystroke)
   const debouncedSetSearchValue = useMemo(
-    () =>
-      debounce(() => {
-        return void refetch();
-      }, 250),
+    () => debounce(setDebouncedSearchValue, 250),
     []
   );
```

Comment:
I'm pretty sure this useMemo is not worthwhile

Reply:
It only prevents the closure from being recreated and adds additional reading complexity that isn't worth it

Reply:
I'm just gonna take a hooks approach and implement debouncing via a useDebounce hook

Reply:
I think what you're doing here is fine, but one of us is confused about what `useMemo()` does

Reply:
I'm pretty sure the main intention is to cache creation of large arrays or data structures when their dependencies don't change, which is not what is happening here.

Reply:
This is probably not the best useMemo use case, useCallback might be good too or even useRef. The main reason of why I went with this is that I saw an article pointing out that the lodash debounce function is expensive and should only be initialized or ran once when the component mounts. I didn't do any performance benchmark to test this tho and this approach requires too much setup. So a simple useDebounce hook might just be what we want

Reply:
Yeah then `useCallback` is the right thing I think